packages/server/postgres/migrations/postDeploy/1620887961308_backfillSegmentId.ts

```diff
@@ -0,0 +1,50 @@
+import {ColumnDefinitions} from 'node-pg-migrate'
+import getRethink from '../../../database/rethinkDriver'
+import getPg from '../../getPg'
+import {backfillSegmentIdQuery} from '../../queries/generated/backfillSegmentIdQuery'
+import catchAndLog from '../../utils/catchAndLog'
+
+export const shorthands: ColumnDefinitions | undefined = undefined
+
+export async function up(): Promise<void> {
+  const r = await getRethink()
+  const batchSize = 3000
+  const doBackfill = async (usersAfterTs?: Date) => {
+    let foundUsers = false
+
+    for (let i = 0; i < 1e5; i++) {
+      console.log('i:', i)
+      const offset = batchSize * i
+      const rethinkUsers = await r
+        .table('User')
+        .between(usersAfterTs ?? r.minval, r.maxval, {index: 'updatedAt'})
+        .orderBy('updatedAt', {index: 'updatedAt'})
+        .filter((row) => row.hasFields('segmentId'))
+        .pluck('id', 'segmentId')
+        .skip(offset)
+        .limit(batchSize)
+        .run()
+      if (!rethinkUsers.length) {
+        break
+      }
+      foundUsers = true
+      const pgUsers = rethinkUsers.map(({id, segmentId}) => ({id, segmentId}))
+      await catchAndLog(() => backfillSegmentIdQuery.run({users: pgUsers}, getPg()))
```

Comment:
Do you have an idea around how many `segmentId`s have been affected? If it's just a couple, it might be worth hard coding the `segmentId <=> id` mapping, so we can avoid a full table scan in rethink.

Reply:
Yeah there will be more than 10K and it's growing every day with newly created accounts 😂

So a full table scan is not avoidable unfortunately.